src/coreComponents/linearAlgebra/utilities/LAIHelperFunctions.hpp

```diff
@@ -157,6 +160,37 @@ void SeparateComponentFilter( MATRIX const & src,
     dst.insert( row, dstIndices.data(), dstValues.data(), k );
   }
   dst.close();
+#else
+
+  array2d< globalIndex > srcIndices( localRows, maxEntries );;
+  array2d< real64 > srcValues( localRows, maxEntries );
+
+  CRSMatrix< real64 > tempMat;
+  tempMat.resize( localRows, src.numGlobalCols(), maxDstEntries );
+
+  for( globalIndex r=0; r<localRows; ++r )
```

Comment:
Did you mean for this to be a parallel kernel launch? Or what's the purpose of using 2D arrays for `srcIndices` and `srcValues`?

Reply:
There are a bunch of host functions in this for loop. They are from hypre, so we don't have control over them. It is pretty strange, but the underlying hypre functions are all host, but if running on device, they take device pointers.